src/OSPSuite.Presentation/Services/DataImporter.cs

```diff
@@ -87,43 +87,41 @@ private static MetaDataCategory createMetaDataCategory<T>(string descriptiveName
       public (IReadOnlyList<DataRepository> DataRepositories, ImporterConfiguration Configuration) ImportDataSets(
          IReadOnlyList<MetaDataCategory> metaDataCategories,
          IReadOnlyList<ColumnInfo> columnInfos,
-         DataImporterSettings dataImporterSettings
+         DataImporterSettings dataImporterSettings,
+         string dataFileName
       )
       {
-         var path = _dialogCreator.AskForFileToOpen(PleaseSelectDataFile, ImportFileFilter, Constants.DirectoryKey.OBSERVED_DATA);
-
-         if (string.IsNullOrEmpty(path))
+         if (string.IsNullOrEmpty(dataFileName))
```

Comment:
Now when the file selection is moved **outside** I would expect that the function is always called with the **valid data file name**.Thus 2 questions/remarks:
* if the `dataFileName` is null/empty - I would throw an exception
* I would also check if the `dataFileName` exists - if not, throw an exception

If throwing exceptions is not appropriated here (because the caller expects different behaviour) then at least the check for `dataFileName` exists should be done (returning the empty data repo array)